src/screens/SendConfirmationSheet.tsx

```diff
@@ -289,6 +301,82 @@ export default function SendConfirmationSheet() {
         ]
   );
 
+  useEffect(() => {
+    if (isENS) {
+      const promises = [
+        estimateGasLimit({
+          address: accountAddress,
+          amount: 0,
+          asset: asset,
+          recipient: toAddress,
+        }),
+      ];
+      const sendENSOptions = Object.fromEntries(
+        checkboxes.map(option => [option.id, option.checked])
+      );
+      if (sendENSOptions['clear-records']) {
+        let records = Object.keys({
+          ...(ensProfile?.data?.coinAddresses || {}),
+          ...(ensProfile?.data?.records || {}),
+        }).reduce((records, recordKey) => {
+          return {
+            ...records,
+            [recordKey]: '',
+          };
+        }, {});
+        if (sendENSOptions['set-address']) {
+          records = { ...records, ETH: toAddress };
+        } else {
+          records = { ...records, ETH: AddressZero };
+        }
+        promises.push(
+          estimateENSSetRecordsGasLimit({
+            name: asset?.name,
+            records: records,
+            ownerAddress: accountAddress,
+          })
+        );
+      } else if (sendENSOptions['set-address']) {
+        promises.push(
+          estimateENSSetAddressGasLimit({
+            name: asset?.name,
+            records: formatRecordsForTransaction({ ETH: toAddress }),
+          })
+        );
+      }
+      if (sendENSOptions['transfer-control']) {
+        promises.push(
+          estimateENSSetOwnerGasLimit({
+            name: asset?.name,
+            ownerAddress: toAddress,
+            fromAddress: accountAddress,
+          })
+        );
+      }
+      promiseUtils
+        .PromiseAllWithFails(promises)
+        .then(gasLimits => {
+          const gasLimit = gasLimits.reduce(
+            (a, b) => parseFloat(a) + parseFloat(b),
```

Comment:
could you use `add` from `utilities`, gasLimits are in wei units so we can have problems parsing them as floats